lib/src/verifiers/utilities/json-schema-verifier.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  JsonSchemaType,
+  jsonTypeSchema
+} from "../../generators/contract/json-schema";
+import { TypeNode } from "../../models/nodes";
+import { DataExpression, DataType, TypeKind } from "../../models/types";
+import Ajv = require("ajv");
+
+export function verifyJsonSchema(
+  dataType: DataType,
+  data: DataExpression,
+  typeStore: TypeNode[]
+) {
+  const ajv = new Ajv();
+  const schema = {
+    ...jsonTypeSchema(dataType),
+    definitions: typeStore.reduce<{ [key: string]: JsonSchemaType }>(
+      (defAcc, typeNode) => {
+        return { [typeNode.name]: jsonTypeSchema(typeNode.type), ...defAcc };
+      },
+      {}
+    )
+  };
+  const validateFn = ajv.compile(schema);
+  const valid = validateFn(dataExpressionToJson(data));
+  if (!valid) {
+    throw new Error(`Invalid: ${ajv.errorsText(validateFn.errors)}`);
+  }
+}
+
+export function dataExpressionToJson(data: DataExpression): any {
```

Comment:
This isn't really JSON Schema specific. What do you think about extracting it out into its own file?

Also, this isn't really generating JSON (the N stands for "Notation"). It's generating JavaScript values. How about calling it `valueFromDataExpression()` or something of the sort?